Modules/sre_lib.h

```diff
@@ -1330,7 +1377,8 @@ SRE(match)(SRE_STATE* state, const SRE_CODE* pattern, int toplevel)
             break;
     }
 
-    return ret; /* should never get here */
+    /* should never get here */
+    Py_UNREACHABLE();
```

Comment:
I'm understanding it again.

Reply:
If `jump == JUMP_NONE`, `SRE(match)` function must be at the toplevel, then `ctx_pos` should be `-1`, it will return before the `swtich` code.
```c
exit:
    ctx_pos = ctx->last_ctx_pos;
    jump = ctx->jump;
    DATA_POP_DISCARD(ctx);
    if (ctx_pos == -1) {   // <- return before swtich code
#ifdef Py_DEBUG
        if (ctx->toplevel && ret >= 0) {
            assert(state->repeat == NULL);
            assert(state->data_stack_base == 0);
        }
#endif
        return ret;
    }
    DATA_LOOKUP_AT(SRE(match_context), ctx, ctx_pos);

    switch (jump) {
        ....
        case JUMP_NONE:
            TRACE(("|%p|%p|RETURN %zd\n", ctx->pattern,
                   ctx->ptr, ret));
            break;
```

Reply:
"SRE(match) function must be at the toplevel", the "toplevel" here means `ctx_pos == -1`.
Sorry, It's been too long.

Reply:
If this line is unreachable, the JUMP_NONE case is unreachable too. You should either put `Py_UNREACHABLE()` just after `case JUMP_NONE:` (maybe even replace `case JUMP_NONE:` with `default:`), or keep lines after `case JUMP_NONE:` and `return ret` just for the case of possible internal errors.

In any case this change is not related to bugs in the RE engine fixed in this PR. I suggest to restore the code and move this change to a separate PR.

Reply:
> In any case this change is not related to bugs in the RE engine fixed in this PR. I suggest to restore the code and move this change to a separate PR.

OK, it's a code polish, not bug fix, reverted.
I'm going to sleep,  solve other problems tomorrow.